proof/invariant-abstract/DetSchedSchedule_AI.thy

```diff
@@ -21173,27 +21173,26 @@ lemma handle_event_valid_sched:
   (* Interrupt *)
   apply wpsimp
       apply (wpsimp wp: handle_interrupt_valid_sched check_budget_restart_valid_sched_weaker)
-     apply (wpsimp wp: check_budget_valid_sched_weaker hoare_vcg_all_lift hoare_vcg_imp_lift')
-    apply(rule_tac Q'="\<lambda>_. valid_sched and invs and ct_not_in_release_q
-                          and ct_in_state activatable
-                          and cur_sc_active
-                          and (\<lambda>s. cur_sc_active s \<longrightarrow> cur_sc_offset_ready (consumed_time s) s)
-                          and consumed_time_bounded
-                          and schact_is_rct and ct_not_queued
-                          and (\<lambda>s. valid_refills (cur_sc s) s)
-                          and current_time_bounded" in hoare_strengthen_post[rotated])
-     apply (clarsimp simp: if_split)
-     apply (intro conjI allI impI)
-           apply fastforce
-          apply (fastforce intro: invs_cur_sc_chargeableE)
-         apply (fastforce elim: ct_in_state_weaken)
-        apply fastforce
-       apply (fastforce intro: invs_cur_sc_chargeableE)
-      apply (fastforce elim: ct_in_state_weaken)
-     apply fastforce
-    apply (wpsimp wp: update_time_stamp_current_time_bounded)
-   apply wpsimp
-   apply (clarsimp simp: ct_in_state_def)
+     apply (wpsimp wp: hoare_drop_imps hoare_vcg_all_lift)
+    apply (wpsimp wp: check_budget_valid_sched_weaker hoare_vcg_all_lift hoare_vcg_imp_lift')
+   apply (rule_tac Q'="\<lambda>_. valid_sched and invs and ct_not_in_release_q
+                           and ct_in_state activatable
+                           and cur_sc_active
+                           and (\<lambda>s. cur_sc_active s \<longrightarrow> cur_sc_offset_ready (consumed_time s) s)
+                           and consumed_time_bounded
+                           and schact_is_rct and ct_not_queued
+                           and (\<lambda>s. valid_refills (cur_sc s) s)
+                           and current_time_bounded" in hoare_strengthen_post[rotated])
+    apply (clarsimp simp: if_split)
+    apply (intro conjI allI impI)
+          apply fastforce
+         apply (fastforce intro: invs_cur_sc_chargeableE)
+        apply (fastforce elim: ct_in_state_weaken)
+       apply fastforce
+      apply (fastforce intro: invs_cur_sc_chargeableE)
+     apply (fastforce elim: ct_in_state_weaken)
+    apply fastforce
```

Comment:
this looks highly compressible, but it also looks like it's not originally yours, so fine to leave